ert_data/obs_loader.py

```diff
@@ -0,0 +1,90 @@
+import pandas as pd
+
+
+def data_loader_factory(observation_type):
+    """
+    Currently, the methods returned by this factory differ. They should not.
+    TODO: Remove discrepancies between returned methods.
+        See https://github.com/equinor/libres/issues/808
+    """
+    if observation_type == "GEN_OBS":
+        return load_general_data
+    elif observation_type == "SUMMARY_OBS":
+        return load_summary_data
+    elif observation_type == "BLOCK_OBS":
+        return load_block_data
+    else:
+        raise TypeError("Unknown observation type: {}".format(observation_type))
+
+
+def load_general_data(facade, observation_key, case_name):
+    observations = facade.get_observations()
+    if observation_key not in observations:
+        return pd.DataFrame()
+
+    obs_vector = observations[observation_key]
+    data_points = []
+    std = []
+    indices = []
+
+    for time_step in obs_vector.getStepList().asList():
+
+        node = obs_vector.getNode(time_step)
+        index_list = [node.getIndex(nr) for nr in range(len(node))]
+
+        data_points.extend(node.get_data_points())
+        std.extend(node.get_std())
+        indices.extend(index_list)
+
+    data = {
+        "OBS": data_points,
+        "STD": std
+    }
+
+    df = pd.DataFrame(data=data, index=index_list).T
+
+    return df
+
+
+def load_block_data(facade, observation_key, case_name):
```

Comment:
case_name not used

Reply:
The file follows the pattern of the loader.py file. The case is required in one case, and not the others, but is there to keep the interface the same for the three functions.